CONTRIBUTING.md

```diff
@@ -28,6 +28,66 @@ use GitHub pull requests for this purpose. Consult
 [GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
 information on using pull requests.
 
+## Pull Request Title Convention
+
+We enforce a PR title convention to quickly indicate type and scope of a PR.
+We also parse it programmatically for changelog generation.
+
+PR titles should
+* be a user facing description of the change
+* be [conventional](https://www.conventionalcommits.org/en/v1.0.0/)
+* append fixed issue(s) at the end (if any)
+
+Examples:
+* `"fix(ui): fixes empty page. Fixes #1234"`
+* `"feat(backend): configurable service account. Fixes #1234, fixes #1235"`
+* `"chore: refactor some files"`
+* `"test: fix CI failure. Part of #1234"` // use "part of" when a PR is an work item of some issue, but shouldn't close the issue when merged.
+
+A quick summary of the conventions we care about:
+### PR Title Structure
+```
+<type>[optional scope]: <description> [Fixes #<issue-number>]
+```
+### PR Type
+Type can be one of
+* feat: A new feature
+* fix: A bug fix (however a bug fix to test infra is not user facing, so it should be test type instead)
+* docs: Documentation only changes
+* chore: Anything else that do not need to be user facing
+* test: Adding missing tests or correcting existing tests
+* refactor: A code change that neither fixes a bug nor adds a feature
+* perf: A code change that improves performance
+
+Note, only feature and fix type PRs will be included in CHANGELOG, because they are
+user facing.
+
+If you think the PR contains multiple types, you can choose the major one or
+split the PR to focused sub-PRs.
+
+If you are not sure which type your PR is and it does not have user impact,
+use `chore` as the fallback.
+
+### PR Scope
+Scope is optional, it can be one of
+* frontend - UI or UI server related, folder `frontend`, `frontend/server`
```

Comment:
Please bold the scope, separate the scope from the description with a colon, and end each list item with w full stop. For example:

* **frontend**: user interface or user interface server related, folder `frontend`, `frontend/server`.

Reply:
UI -> user interface